rust_snuba/src/factory_v2.rs

```diff
@@ -0,0 +1,283 @@
+use sentry_arroyo::processing::strategies::healthcheck::HealthCheck;
+use sentry_arroyo::processing::strategies::reduce::Reduce;
+use sentry_arroyo::processing::strategies::run_task_in_threads::{
+    ConcurrencyConfig, RunTaskInThreads,
+};
+use sentry_arroyo::processing::strategies::run_task_in_threads::{
+    RunTaskError, RunTaskFunc, TaskRunner,
+};
+use sentry_arroyo::processing::strategies::{ProcessingStrategy, ProcessingStrategyFactory};
+use sentry_arroyo::types::Message;
+use sentry_arroyo::types::{Partition, Topic};
+use sentry_kafka_schemas::Schema;
+
+use crate::config;
+use crate::metrics::global_tags::set_global_tag;
+use crate::processors::{self, get_cogs_label};
+use crate::strategies::accountant::RecordCogs;
+use crate::strategies::clickhouse::batch::{BatchFactory, HttpBatch};
+use crate::strategies::clickhouse::ClickhouseWriterStep;
+use crate::strategies::commit_log::ProduceCommitLog;
+use crate::strategies::healthcheck::HealthCheck as SnubaHealthCheck;
+use crate::strategies::join_timeout::SetJoinTimeout;
+use crate::strategies::processor::{
+    get_schema, make_rust_processor, make_rust_processor_with_replacements, validate_schema,
+};
+use crate::strategies::python::PythonTransformStep;
+use crate::strategies::replacements::ProduceReplacements;
+use crate::types::{BytesInsertBatch, CogsData, RowData};
+
+pub struct ConsumerStrategyFactoryV2 {
+    pub storage_config: config::StorageConfig,
+    pub env_config: config::EnvConfig,
+    pub logical_topic_name: String,
+    pub max_batch_size: usize,
+    pub max_batch_time: Duration,
+    pub processing_concurrency: ConcurrencyConfig,
+    pub clickhouse_concurrency: ConcurrencyConfig,
+    pub commitlog_concurrency: ConcurrencyConfig,
+    pub replacements_concurrency: ConcurrencyConfig,
+    pub async_inserts: bool,
+    pub python_max_queue_depth: Option<usize>,
+    pub use_rust_processor: bool,
+    pub health_check_file: Option<String>,
+    pub enforce_schema: bool,
+    pub commit_log_producer: Option<(Arc<KafkaProducer>, Topic)>,
+    pub replacements_config: Option<(KafkaConfig, Topic)>,
+    pub physical_consumer_group: String,
+    pub physical_topic_name: Topic,
+    pub accountant_topic_config: config::TopicConfig,
+    pub stop_at_timestamp: Option<i64>,
+    pub batch_write_timeout: Option<Duration>,
+    pub custom_envoy_request_timeout: Option<u64>,
+    pub join_timeout_ms: Option<u64>,
+    pub health_check: String,
+}
+
+impl ProcessingStrategyFactory<KafkaPayload> for ConsumerStrategyFactoryV2 {
+    fn update_partitions(&self, partitions: &HashMap<Partition, u64>) {
+        match partitions.keys().map(|partition| partition.index).min() {
+            Some(min) => set_global_tag("min_partition".to_owned(), min.to_string()),
+            None => set_global_tag("min_partition".to_owned(), "none".to_owned()),
+        }
+    }
+
+    fn create(&self) -> Box<dyn ProcessingStrategy<KafkaPayload>> {
```

Comment:
nit: This function has a lot of code. Consider refactoring using smaller methods.